Resources/doc/reference/configuration/user-context.rst

```diff
@@ -120,6 +120,16 @@ request. However, when you decide to cache hash responses, you must invalidate
 them when the user context changes, particularly when the user logs in or out.
 This bundle provides a logout handler that takes care of this for you.
 
+.. _always_vary_on_context_hash:
+
+``always_vary_on_context_hash``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: ``true``
+
+Set ``always_vary_on_context_hash`` to  ``false`` to disable always adding X-User-Context-Hash or Cookie, Authorization in Vary header.
```

Comment:
Well with an ezPublish web app and "ez customized" Varnish vcl, there is always a User Hash in the request. So the listener will only add Vary Cookie, Authorization when going through Apache directly. 
But in a more general sense, I think adding a Vary Cookie to cache different responses in Varnish depending on cookie header value (which will ALWAYS be different for any website user, especially if you set a session cookie) is a not a very good practice and can be dangerous regarding Varnish cache size which could be very quickly overloaded. That's why I decided to completely skip the Vary modification with this option. 
And for those who want to keep the Vary Cookie, Authorization it will not be a problem as it will remain the default behavior when using the http cache bundle.